sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -76,6 +77,23 @@ public void verbose(String message) {
         }
     }
 
+    /**
+     * Logs a message at {@code verbose} log level.
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Logging a message at verbose log level.</p>
+     *
+     * {@codesnippet com.azure.core.util.logging.clientlogger.deferredVerbose#supplier}
+     *
+     * @param message The message to log.
+     */
+    public void deferredVerbose(Supplier<String> message) {
```

Comment:
We will also need to support passing in an exception and depending on the log level, we determine whether to log the stacktrace of the exception or not. We will need another overload for accepting a `Throwable` for each of the log levels. This results in too many methods. 

So, it might be better to just add the following 2 methods instead:

```java
void log(LogLevel level, Supplier<String> message);
void log(LogLevel level, Supplier<String> message, Throwable throwable);
```

Reply:
If it is needed, I can add these at the end of next week

Reply:
Yeah, it would be great if this change can be made! Next week sounds good, thank you!

Reply:
Are we sure, we need the 2nd one? void log(LogLevel level, Supplier<String> message, Throwable throwable); In this case we already have a message, from throwable.getMessage() so I'm not sure the purpose of this one.  Or should I concatenate the string from the supplier and from the throwable.getMessage()?

Reply:
In `ClientLogger`, log methods take varargs and if the last argument is an instance of `Throwable` and if the user has set the log level to `verbose`, we additionally log the stacktrace from the exception. Otherwise, just the message is logged. So, the 2nd overload is to enable logging the stacktrace, if necessary.
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java#L323-L329

Reply:
Makes sense, thanks!

Reply:
I have implemented the changes. Locally (on arch linux) it is working fine, also the ubuntu 18 111, but the  java - core - ci (Build Test ubuntu1804_18) is failing.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=885563&view=logs&j=427277b2-9330-5c1d-39b6-f7ca00c2b97d&t=0dd1d86a-7289-506f-2b00-71cf0f91bdc8
Is there a way, I can retrigger this part manually?

Reply:
I have restarted the failed builds.